guides/Benchmarking_shuffle_guide.md

```diff
@@ -1,11 +1,22 @@
 # OUTLINE
+
 ## Reasoning for benchmarking models created across DLC 3.0 and earlier versions
 
-DLC 3.0 runs on PyTorch as the engine rather than TensorFlow. It is of importance for replicability of data analysis to benchmark existing models created using DLC versions prior to 3.0 against new models created in DLC 3.0 and later versions.
+DLC 3.0 runs on PyTorch as the engine rather than TensorFlow. It is of importance for
+replicability of data analysis to benchmark existing models created using DLC versions
+prior to 3.0 against new models created in DLC 3.0 and later versions.
 
-When benchmarking different models, maintaining the same test-train data split is crucial for ensuring comparability. If the models use differing train and test datasets, their performance metrics cannot be accurately compared because they are not trained on the same data. This is especially important when comparing the performance of different models, such as a TensorFlow model and a PyTorch model, or two models with different architectures. Using the same training set is necessary to ensure fair comparisons, as different training sets may yield different results and make it difficult to accurately compare the models' performance.
+When benchmarking different models, maintaining the same test-train data split is 
+crucial for ensuring comparability. If the models use differing train and test datasets,
+their performance metrics cannot be accurately compared because they are not trained on
+the same data. This is especially important when comparing the performance of different
+models, such as a TensorFlow model and a PyTorch model, or two models with different
+architectures. Using the same training set is necessary to ensure fair comparisons, as
+different training sets may yield different results and make it difficult to accurately
+compare the models' performance.
```

Comment:
>  If the models use differing train and test datasets, their performance metrics cannot be accurately compared because they are not trained on the same data.

I'd go a bit more in depth here, especially for the fixed test set - I think we can emphasize:
- Comparing metrics on different test sets does not make sense. Take the simple case where there's a single test image. If we compare the RMSE of a model on an "easy" image against the RMSE of another model on a "hard" image, that doesn't actually tell us which model is better. We need to compare the errors the models make on the same images. 

> Using the same training set is necessary to ensure fair comparisons, as different training sets may yield different results and make it difficult to accurately compare the models' performance.

Here I would maybe emphasize more that the model performance cannot be compared: we don't know if one model is better than the other because the architecture performs better, or if it's because the images it was trained on were "easier"/"better" to learn from - we want to "decouple" the dataset from the model architecture.

Reply:
I think that's very good - one thing is I would split

> For example, if we compare the RMSE of a model on an "easy" test image with the RMSE of another model on a "hard" test image, it doesn't determine whether a model is better than the other because the architecture performs better or because the training images were "better" to learn from.

into two sentences (one saying we need to compare metrics computed on the same test images, and one saying we need to train the models on the same images to "decouple" the dataset from the architecture)

Reply:
How is this ?
```suggestion


- When comparing different models, it's important to use the same train-test data split to ensure fair comparisons. If the models are trained on different datasets, their performance metrics can't be accurately compared. This is crucial when comparing the performance of models with different architectures or different sets of hyperparameters. For example, if we compare the RMSE of a model on an "easy" test image with the RMSE of another model on a "hard" test image, it doesn't determine whether a model is better than the other because the architecture performs better or because the training images were "better" to learn from. Thus, we not only need to compare the models based on metrics computed on the same test images, but also train them on an indentical fixed training set  in order to "decouple" the dataset from the model architecture.
```

Reply:
Good with me! Once this is updated, I think we can merge this into the main branch.

Then we can port the pull request to the main DeepLabCut repository

Reply:
Yes, all changes have been added!